src/net/mod.rs

```diff
@@ -49,7 +49,7 @@ pub struct NetworkUsers<S: Default + Miniconf + Clone, T: Serialize> {
     pub processor: NetworkProcessor,
     stream: DataStream,
     generator: Option<FrameGenerator>,
-    pub telemetry: TelemetryClient<T>,
+    pub telemetry: Option<TelemetryClient<T>>,
```

Comment:
This is fundamentally unsound to do because it allows the user to move the client out of `NetworkUsers`. The `TelemetryClient` has to remain in this structure otherwise it's possible that the network stack could be used in multiple contexts and would result in pre-emption.

Reply:
I see! And totally agree this should not be done.

Reply:
The purpose of this change was only to stop the `NetworkUsers::update()` function from polling the telemetry's MQTT client. Splitting the API into `update_stream()`, `update_miniconf()` and `update_telemetry()` achieves the same goal. `update()` (or more explicitly `update_all()`) could then do all the above as usual.